swap/src/alice.rs

```diff
@@ -82,10 +81,7 @@ pub async fn swap(
     );
     swarm.set_state0(state0.clone());
 
-    let state1 = match message0 {
-        Some(msg) => state0.receive(msg).expect("failed to receive msg 0"),
-        None => panic!("should have the message by here"),
-    };
+    let state1 = state0.receive(message0).expect("failed to receive msg 0");
```

Comment:
I don't understand how the compiler knows that this variable was defined... 

//edit// ah, because you only break the loop in a single case in which this variable is defined... very interesting.